### DISCUSSION TITLE
Turn the bookmark system into an actual chaptering feature

### DISCUSSION THREAD

**Discussion by Mechanizen** [ppy/osu#23154] [Ideas]
Similarly to Clone Hero where you can give names to sections/solos etc... I think it would be a great addiction to be able to name sections or a map. 

From the player side it makes it easier to talk about certain parts of a map instead of just saying "the ending is hard", "I can't do this part" or "this section is x". 

From the mapper point of view it would also make it easier for feedback/modding if we can give names to specific sections. For quick feedback, instead of having to give a timestamp interval saying like "from 00:10:486(1) - to 00:32:486(1)" we would be able to refer to the name of the section if there is one. So we could just say "I think Slow2 is overmapped" or "Solo1 is underwhelming" instead of copypasting timestamps which don't mean much at first glance.

Naming sections would also give mappers a new tool to express their vision of the song/their map. What I'm thinking about is for instance Soulless 5 on Clone Hero where the maker named some sections like "Caged" or "Pirates", when talking about those names to someone who knows the chart, they will know instantly what you are talking about.

Storyboard elements with the names of the section could then be automatically generated, allowing mappers who don't really have experience in storyboarding to add some context to their maps for presentation videos/etc... It would also be very useful to plan collabs.

Also pushing a bit further, it could even be used to create "content tables", allowing the user to browse through a map. For modding it would be possible to copypaste the timestamp of the begining of the section and have the name of the section displayed in the mod instead of the raw timestamps. (Example: section starts at 00:10:486, I copy the timestamp '00:10:486' and paste it to my mod. As this timestamp does not refer to a specific object, it is recognised as a 'section timestamps' and so it is displayed as 'Section name' in my mod instead of just a raw '00:10:486' timestamp)

The section naming could be added by simply allowing us to name bookmarks. The sole purpose of bookmarks has always been to draw sections on a map but the feature never went beyond just drawing blue lines on the timeline. 

I think being able to name bookmarks and thus name sections is a logic development for this feature that could lead to healthy results.

**TLDR**: allow us to give names to bookmarks to turn the bookmark system into an actual chaptering system. Allowing for improved communication for players, mappers and modders and opening the gate for more feature development.

**Comment by Rekunan** [comment]
Duplicate of #12978
